core/lib/network-recorder.js

```diff
@@ -306,6 +286,33 @@ class NetworkRecorder extends RequestEventEmitter {
     // get out the list of records & filter out invalid records
     const records = networkRecorder.getRawRecords().filter(record => record.isValid);
 
+    // Set record.source to the Crdp target type that requested the resource,
+    // based on the what session the network events came from.
+    {
+      /** @type {Map<string, 'page'|'iframe'|'worker'>} */
+      const sessionIdToTargetType = new Map();
+      for (const message of devtoolsLog) {
+        if (message.method === 'Target.attachedToTarget') {
+          if (['page', 'iframe', 'worker'].includes(message.params.targetInfo.type)) {
+            // @ts-expect-error type narrowed.
+            sessionIdToTargetType.set(message.params.sessionId, message.params.targetInfo.type);
+          }
+        }
+      }
+
+      const requestIdToSessionId = new Map();
```

Comment:
Seems like we only need this map because the `sessionId` isn't on the network request anymore. WDYT about just keeping the `sessionId` on the network request (but still remove all that `_mainSessionId` BS)?

Reply:
+1 to keeping sessionId on the network request. Can't depend on them acting the same across platforms but useful when e.g. matching them back to targets when debugging